ops/model.py

```diff
@@ -1047,25 +1047,25 @@ def stop(self, *service_names: str):
         """Stop given service(s) by name."""
         self._pebble.stop_services(service_names)
 
-    def merge_layer(self, layer: typing.Union[str, typing.Dict, 'pebble.Layer']):
-        """Dynamically merge layer onto the Pebble setup layers.
-
-        Pebble merges the layer with the existing dynamic layer according to
-        its flattening/override rules, or adds a new dynamic if there are no
-        dynamic layers.
+    def add_layer(self, label: str, layer: typing.Union[str, typing.Dict, 'pebble.Layer'],
+                  combine: bool = False):
+        """Dynamically add a new layer onto the Pebble configuration layers.
 
         Args:
-            layer: A YAML string, setup layer dict, or pebble.Layer object
-                containing the Pebble layer to add.
+            label: Label for new layer (and label of layer to merge with if
```

Comment:
I'm guessing the label has to be unique unless "combine: true"? Otherwise that makes combine ambiguous.

Reply:
That's what I was thinking, but then I forgot to bake this assumption into `AppendLayer`. Now fixed! https://github.com/canonical/pebble/pull/10/commits/625783a5cf24ebbf2f398a19a3c6862c7a065eb7